modules/rostests/kmtests/kmtest/service.c

```diff
@@ -227,6 +246,28 @@ KmtStopService(
     if (!ControlService(*ServiceHandle, SERVICE_CONTROL_STOP, &ServiceStatus))
         error_goto(Error, cleanup);
 
+    ServiceNameOut = ServiceName ? ServiceName : L"(handle only, no name)";
+
+    if (ServiceStatus.dwCurrentState == SERVICE_STOPPED)
+    {
+        trace("Service %ls state = STOPPED, immediately\n", ServiceNameOut);
+        goto cleanup;
+    }
+
+    // Give up timeslice, so the service has a chance to actually stop.
+    // If this is not enough, then probably increase to 1000 (= 1 second)...
+    Sleep(0);
```

Comment:
This is just asking for a race condition. You need a wait loop for it to be reliable.

Reply:
Done.